features/nanostack/nanostack-interface/Nanostack.cpp

```diff
@@ -534,6 +586,74 @@ const char *Nanostack::get_ip_address()
     return "::";
 }
 
+nsapi_error_t Nanostack::gethostbyname(const char *name, SocketAddress *address, nsapi_version_t version, const char *interface_name)
+{
+    if (name[0] == '\0') {
+        return NSAPI_ERROR_PARAMETER;
+    }
+    // check for simple ip addresses
+    if (address->set_ip_address(name)) {
+        if (version != NSAPI_UNSPEC && address->get_ip_version() != version) {
+            return NSAPI_ERROR_DNS_FAILURE;
+        }
+        return NSAPI_ERROR_OK;
+    }
+
+    // try nanostack DNS cache, if not found then fallback to dns query
+    if (nanostack_dns_query_result_check(name, address, interface_name) == 0) {
+        return 0;
+    }
+
```

Comment:
Should have the "if (version == NSAPI_UNSPEC) { " and in that case set version ipv6. Resolving ipv4 addresses with nanostack is bit weird since stack does not have ipv4, but I guess no reason to force reject for those either.